Control/Monad/Logic.hs

```diff
@@ -83,7 +83,7 @@ observeT :: MonadFail m => LogicT m a -> m a
 observeT lt = unLogicT lt (const . return) (fail "No answer.")
 
 -------------------------------------------------------------------------
--- | Extracts all results from a LogicT computation.
+-- | Tries to extract all results from a LogicT computation.
```

Comment:
Sorry, but I don't think this is a right wording. "Tries" implies that there is some notion of failure, like returning `Maybe (m [a])`, which is not the case. 

I think it would be better to include examples you came up with in #10 to haddocks.

Reply:
no need to apologize :)

How would you feel about
"Extracts all results from a LogicT computation, but may blocked by the underlying monad"
?

I'll try to clean up those examples to add to the pull request.

Reply:
This sounds better. It would be desirable to explain when `observeManyT 10` is productive, but `take 10 . observeAll` is not.